classes/query/Query.php

```diff
@@ -3,142 +3,138 @@
 /**
  * @file classes/query/Query.php
  *
- * Copyright (c) 2016-2024 Simon Fraser University
- * Copyright (c) 2003-2024 John Willinsky
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
  * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
  *
  * @class Query
  *
- * @ingroup submission
- *
- * @see QueryDAO
- *
  * @brief Class for Query.
  */
 
 namespace PKP\query;
 
-use Illuminate\Support\LazyCollection;
-use PKP\core\PKPApplication;
-use PKP\db\DAORegistry;
-use PKP\db\DAO;
-use PKP\note\Note;
+use APP\facades\Repo;
+use Eloquence\Behaviours\HasCamelCasing;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasMany;
 
-class Query extends \PKP\core\DataObject
+class Query extends Model
 {
-    /**
-     * Get query assoc type
-     *
-     * @return int Application::ASSOC_TYPE_...
-     */
-    public function getAssocType()
-    {
-        return $this->getData('assocType');
-    }
+    use HasCamelCasing;
 
-    /**
-     * Set query assoc type
-     *
-     * @param int $assocType Application::ASSOC_TYPE_...
-     */
-    public function setAssocType($assocType)
+    const CREATED_AT = 'date_posted';
+    const UPDATED_AT = 'date_modified';
+
+	protected $table = 'queries';
+	protected $primaryKey = 'query_id';
+
+    protected $fillable = [
+        'assocType', 'assocId', 'stageId', 'seq',
+        'datePosted', 'dateModified', 'closed'
+    ];
+
+    protected function casts(): array
```

Comment:
In eloquent models, attribute casting should be declared by using $casts property, not a method.
```
protected $casts = [
    'assocType' => 'int',
    'assocId' => 'int',
    'stageId' => 'int',
    'seq' => 'float',
    'datePosted' => 'datetime',
    'dateModified' => 'datetime',
    'closed' => 'boolean',
];
```